source/common/upstream/health_checker_base_impl.cc

```diff
@@ -34,13 +34,28 @@ HealthCheckerImplBase::HealthCheckerImplBase(const Cluster& cluster,
       unhealthy_edge_interval_(
           PROTOBUF_GET_MS_OR_DEFAULT(config, unhealthy_edge_interval, unhealthy_interval_.count())),
       healthy_edge_interval_(
-          PROTOBUF_GET_MS_OR_DEFAULT(config, healthy_edge_interval, interval_.count())) {
+          PROTOBUF_GET_MS_OR_DEFAULT(config, healthy_edge_interval, interval_.count())),
+      transport_socket_options_(initTransportSocketOptions(config)) {
   cluster_.prioritySet().addMemberUpdateCb(
       [this](const HostVector& hosts_added, const HostVector& hosts_removed) -> void {
         onClusterMemberUpdate(hosts_added, hosts_removed);
       });
 }
 
+std::shared_ptr<Network::TransportSocketOptionsImpl>
+HealthCheckerImplBase::initTransportSocketOptions(
+    const envoy::api::v2::core::HealthCheck& config) const {
+  if (config.has_tls_options()) {
+    std::vector<std::string> protocols{config.tls_options().alpn_protocols().begin(),
+                                       config.tls_options().alpn_protocols().end()};
+    return std::make_shared<Network::TransportSocketOptionsImpl>("", std::vector<std::string>{},
+                                                                 std::move(protocols));
+  }
+
+  return std::make_shared<Network::TransportSocketOptionsImpl>("", std::vector<std::string>{},
+                                                               std::vector<std::string>{});
```

Comment:
nit: I think just `return std::maked_shared<Network::TransportSocketOptionsImpl>();` should suffice since all the parameters have default values equivalent to what you're providing here.

https://github.com/envoyproxy/envoy/blob/c9af58c02cd4144a811a1a18a1539cb3710af019/source/common/network/transport_socket_options_impl.h#L11